client/state/audio/middleware.js

```diff
@@ -0,0 +1,62 @@
+/**
+ * External dependencies
+ */
+import { get } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import {
+	HAPPYCHAT_RECEIVE_EVENT,
+} from 'state/action-types';
+
+// Maps the audio reference to the audio file source.
+// In future we might want to store this in Redux if
+// we want the sounds to be configurable.
+const REFERENCE_MAP = {
+	'happychat-message-received': '/calypso/audio/chat-pling.wav',
+};
+
+const getSourceForReference = reference => {
+	return get( REFERENCE_MAP, reference );
+};
+
+export const playSound = reference => {
+	if ( typeof Audio !== 'function' ) {
+		// No Audio support in this browser
+		return;
+	}
+
+	const audioSource = getSourceForReference( reference );
+	if ( audioSource == null ) {
+		return;
+	}
+
+	const audioClip = new Audio( audioSource );
+	audioClip.play();
+};
+
+export const onHappyChatMessage = ( dispatch, { event } ) => {
+	event.source !== 'customer' && playSound( 'happychat-message-received' );
```

Comment:
is having a map necessary since these references and files all appear to be contained in this single file? maybe with the added simplicity of having all the sounds tied to the actions in one place the mapping isn't as important, or at least, with only a single sound it isn't import yet?